pkg/gohai/filesystem/filesystem_nix.go

```diff
@@ -0,0 +1,196 @@
+// This file is licensed under the MIT License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2014-present Datadog, Inc.
+
+//go:build darwin || linux
+// +build darwin linux
+
+package filesystem
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/sys/unix"
+
+	log "github.com/cihub/seelog"
+	"github.com/moby/sys/mountinfo"
+)
+
+// These FS types should be excluded from listing
+var IgnoredFSTypes = map[string]struct{}{
+	"autofs":      {},
+	"debugfs":     {},
+	"devfs":       {},
+	"devpts":      {},
+	"devtmpfs":    {},
+	"fuse.portal": {},
+	"fusectl":     {},
+	"ignore":      {},
+	"kernfs":      {},
+	"mqueue":      {},
+	"none":        {},
+	"proc":        {},
+	"rpc_pipefs":  {},
+	"squashfs":    {},
+	"subfs":       {},
+	"sysfs":       {},
+}
+
+// These FS types are known to be remote
+var RemoteFSTypes = map[string]struct{}{
+	"acfs":       {},
+	"afs":        {},
+	"auristorfs": {},
+	"coda":       {},
+	"fhgfs":      {},
+	"gpfs":       {},
+	"ibrix":      {},
+	"ocfs2":      {},
+	"vxfs":       {},
+}
+
+// FSInfoGetter provides function to get information about a given filesystem: its size and its dev id
+type FSInfoGetter interface {
+	// Size returns the size of the given filesystem
+	Size(mount *mountinfo.Info) (uint64, error)
+
+	// Dev returns the dev id of the given filesystem
+	// the return type is interface{} because `syscall.Stat_t` uses different types for Dev depending on the platform
+	Dev(mount *mountinfo.Info) (interface{}, error)
+}
+
+type UnixFSInfo struct{}
+
+func (UnixFSInfo) Size(mount *mountinfo.Info) (uint64, error) {
+	var statfs unix.Statfs_t
+	if err := unix.Statfs(mount.Mountpoint, &statfs); err != nil {
+		return 0, fmt.Errorf("statfs %s: %v", mount.Source, err)
+	}
+
+	sizeKB := statfs.Blocks * uint64(statfs.Bsize) / 1024
+	return sizeKB, nil
+}
+
+func (UnixFSInfo) Dev(mount *mountinfo.Info) (interface{}, error) {
```

Comment:
As explained in the definition of the interface:
> 	// Dev returns the dev id of the given filesystem
>	// the return type is interface{} because `syscall.Stat_t` uses different types for Dev depending on the platform

The type of `syscall.Stat_t` is `int32` on Darwin and `uint64` on Linux.
I guess I could always return uint64 but I'm worried the go linter would trigger on Linux because the cast is useless :'(
But at worst I can use a //nolint

Reply:
Seems like even just on Darwin the type could be different (from the man page of stat (2) ):
> COMPATIBILITY
     Previous versions of the system used different types for the st_dev, st_uid, st_gid, st_rdev, st_size, st_blksize and st_blocks fields.

Reply:
Done in 53f2e4e400562ad092b168e009cf7831862fc453